internal/services/mssql/mssql_managed_instance_resource_test.go

```diff
@@ -1233,6 +1233,15 @@ func (r MsSqlManagedInstanceResource) Exists(ctx context.Context, client *client
 
 func (r MsSqlManagedInstanceResource) basic(data acceptance.TestData) string {
 	return fmt.Sprintf(`
+
+provider "azurerm" {
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
+    }
+  }
+}
```

Comment:
this is an indication of a bug where the delete is not waiting long enough for the resources to be fully removed/cleaned up? ie users will encounter this same issue?

Reply:
Hi @katbyte - I think this is a problem with `ListResourcesForResourceGroup ` because I added test code in func `resourceResourceGroupDelete` to test whether virtualCluster really exists when the resource group is being deleted. The result is that virtualCluster  does not exist at that time, but the result returned by ListResourcesForResourceGroup it that virtualCluster still exists. Whereas the [comment](https://github.com/hashicorp/terraform-provider-azurerm/blob/dff425415de4b74f7f156c882cc8ed2b3977a05c/internal/services/resource/resource_group_resource.go#L133) also indicate that "resource groups sometimes hold on to resource information after the resources have been deleted" and the test case does not always fail with this error. So I think it would be appropriate to add `prevent_deletion_if_contains_resources ` to resolve test failures. WDYT?

For details, set the test code below and  log in this [runner](https://ci-oss.hashicorp.engineering/buildConfiguration/TerraformOpenSource_TerraformProviders_AzureRMPublic_AZURERM_SERVICE_PUBLIC_MSSQL/384245).
![image](https://user-images.githubusercontent.com/39109137/231992734-2f2f98a0-2b96-4fa3-8336-befa10953dfc.png)